tpot/builtins/embedding_estimator.py

```diff
@@ -0,0 +1,143 @@
+    """Meta-transformer for creating neural network embeddings as features.
+    """
+
+    def __init__(self, estimator, embedding_layer=None, backend=None):
+        """Create a StackingEstimator object.
+
+        Parameters
+        ----------
+        estimator: neural network model; either from sklearn or Keras-like.
+            The estimator to generate embeddings.
+        embedding_layer: the particular layer used as the embedding. 
+            By default we use the second last layer. Layers are counted with
+            input layer being `0th` layer; negative indices are allowed. 
+        backend: (optional), the backend we use to query the neural network. 
+            Not required if using scikit-learn interface.
+            Currently only supports keras-like interface (incl. tensorflow)
+        """
+        second_last_layer = -2
+        self.estimator = estimator
+        self.embedding_layer = (
+            second_last_layer if embedding_layer is None else embedding_layer
+        )
+        self.backend = backend
+
+    def fit(self, X, y=None, **fit_params):
+        """Fit the StackingEstimator meta-transformer.
+
+        Parameters
+        ----------
+        X: array-like of shape (n_samples, n_features)
+            The training input samples.
+        y: array-like, shape (n_samples,)
+            The target values (integers that correspond to classes in classification, real numbers in regression).
+        fit_params:
+            Other estimator-specific parameters.
+
+        Returns
+        -------
+        self: object
+            Returns a copy of the estimator
+        """
+        if not issubclass(self.estimator.__class__, MLPClassifier) and not issubclass(
+            self.estimator.__class__, MLPRegressor
+        ):
+            input_shape = X.shape[1]
+            self.estimator.sk_params["input_shape"] = input_shape
+            self.estimator.check_params(self.estimator.sk_params)
+        self.estimator.fit(X, y, **fit_params)
+        return self
+
+    def transform(self, X):
+        """Transform data by adding embedding as features.
+
+        Parameters
+        ----------
+        X: numpy ndarray, {n_samples, n_components}
+            New data, where n_samples is the number of samples and n_components is the number of components.
+
+        Returns
+        -------
+        X_transformed: array-like, shape (n_samples, n_features + embedding) where embedding is the size of the embedding layer
+            The transformed feature set.
+        """
+        X = check_array(X)
+        X_transformed = np.copy(X)
+        # add class probabilities as a synthetic feature
+        if issubclass(self.estimator.__class__, MLPClassifier) or issubclass(
+            self.estimator.__class__, MLPRegressor
+        ):
+            X_transformed = np.hstack(
+                (self._embedding_mlp(self.estimator, X), X_transformed)
+            )
+        else:
```

Comment:
Chose not to do checking here otherwise you'll need to do an import at the top; which adds whatever flavour of the month neural network library to be a hard dependency to this. I don't have better ideas at this point but happy for any suggestions